### PR TITLE
Battle net oauth2

### PR DESCRIPTION
**Pull request by gvlasov**
Continues on https://github.com/go-pkgz/auth/pull/53

Add Battle.net auth

Depends on latest commit in https://github.com/go-pkgz/auth, so needs changes

### CODE REVIEW

**Comment by umputun** [comment]
What is the use case for such an exotic authentication provider in remark42? I'm trying to keep the list of providers limited to the popular options only.

**Comment by gvlasov** [comment]
@umputun Battle.net has 30+ million active monthly users, and it is used for auth in many social apps. If that is not popular enough, lets close this PR, my intention here is just to push upstream what I have implemented in my project so I don't have to maintain a fork.

**Comment by paskal** [comment]
@gvlasov do you have a particular site you want to use it in? E.g. you prepared a PR because you were able to add it or because you need it for particular site of yours?

**Comment by gvlasov** [comment]
@paskal > you prepared a PR because you were able to add it or because you need it for particular site of yours

Both.

**Comment by umputun** [comment]
> Battle.net has 30+ million active monthly users, and it is used for auth in many social apps

I have never seen any popular site/system offering Battle.net auth as an option. In fact, I have never seen any site offering such an option. Maybe it is just me, not sure. The point I'm making - is it really "used for auth in many social apps" outside of the gaming community? 

I have also checked if auth0 provides any connectors for Battle.net and in the long list of their "social connections" I have not seen Battle.net.

**Comment by gvlasov** [comment]
@umputun 
> is it really "used for auth in many social apps" outside of the gaming community?

Outside of gaming community - sure not, it is a gaming platform after all. It is mostly used in various third party player statistics services like https://www.overbuff.com/ or https://overwatchtracker.com/, as well as on the official websites.

File: README.md [hunk 1]
```diff
@@ -646,7 +662,7 @@ type User struct {
 }
 ```
 
-_currently supported providers are `google`, `facebook`, `github` and `yandex`_
+_currently supported providers are `google`, `facebook`, `github`, `yandex`and `battlenet`
```
Comment (paskal) [inline README.md:665]: missing closing `_`

File: ode [hunk 2]
```diff
@@ -0,0 +1,4 @@
+# Netscape HTTP Cookie File
```
Comment (paskal) [inline ode:1]: Is there a reason to commit this file?

File: backend/vendor/modules.txt [hunk 3]
```diff
@@ -72,7 +72,7 @@ github.com/go-chi/cors
 # github.com/go-chi/render v1.0.1
 ## explicit
 github.com/go-chi/render
-# github.com/go-pkgz/auth v0.10.2
+# github.com/go-pkgz/auth v0.10.3-0.20200615230206-4150d016cbbf
```
Comment (paskal) [inline backend/vendor/modules.txt:75]: `0.11.0` should have what you are looking for.

Comment (paskal) [reply backend/vendor/modules.txt:75]: Actually, it's already in this repo so the update is no longer necessary.

File: Dockerfile [hunk 4]
```diff
@@ -77,4 +77,10 @@ HEALTHCHECK --interval=30s --timeout=3s CMD curl --fail http://localhost:8080/pi
 
 COPY docker-init.sh /srv/init.sh
 RUN chmod +x /srv/init.sh
+COPY backend/app/cmd/testdata/cert.pem /cert.pem
```
Comment (paskal) [inline Dockerfile:80]: Is it related to battle.net oauth? Should it be present in production docker image? This doesn't seem right in this PR and at least deserves a comment in code (in Dockerfile).

File: ode (hunk 2 above)
Comment (gvlasov) [reply ode]: @paskal Removed the file, added by a mistake

File: README.md (hunk 1 above)
Comment (gvlasov) [reply README.md]: @paskal Fixed

File: backend/vendor/modules.txt (hunk 3 above)
Comment (gvlasov) [reply backend/vendor/modules.txt]: @paskal Reverted to the version in master

File: Dockerfile (hunk 4 above)
Comment (gvlasov) [reply Dockerfile]: @paskal Nope, not related, these are leftovers from local testing. Removed

**Pull request closed without merging**